src/pipeline_migration/cli.py

```diff
@@ -16,17 +17,24 @@
 logging.basicConfig(level=logging.DEBUG, format="%(levelname)s:%(asctime)s:%(name)s:%(message)s")
 logger = logging.getLogger("cli")
 
-SCHEMA_UPGRADE: Final = {
+SCHEMA_UPGRADE: Final[dict[str, Any]] = {
+    "$schema": "https://json-schema.org/draft/2020-12",
+    "title": "Schema for Renovate upgrade data",
     "type": "object",
     "properties": {
-        "depName": {"type": "string"},
-        "currentValue": {"type": "string"},
+        "depName": {"type": "string", "minLength": 1},
+        "currentValue": {"type": "string", "minLength": 1},
         "currentDigest": {"type": "string", "pattern": "^sha256:[0-9a-f]+$"},
-        "newValue": {"type": "string"},
+        "newValue": {"type": "string", "minLength": 1},
         "newDigest": {"type": "string", "pattern": "^sha256:[0-9a-f]+$"},
-        "depTypes": {"type": "array", "items": {"type": "string"}},
-        "packageFile": {"type": "string"},
-        "parentDir": {"type": "string"},
+        "depTypes": {
+            "type": "array",
+            "minItems": 1,
+            "items": {"type": "string"},
+            "contains": {"const": "tekton-bundle"},
+        },
```

Comment:
I'm not sure how I feel about using schema validation to check that the `depTypes` array contains a specific element. The upgrade object isn't _invalid_ if it doesn't contain the element, we just don't care about it. And I feel that should be done in the code, not in the schema. The way it was done before (`if "tekton-bundle" not in upgrade["depTypes"]`) made more sense to me.

Reply:
All of these comments are addressed. PTAL.